api/v1beta1/openstackdataplane_types.go

```diff
@@ -28,11 +28,8 @@ import (
 type OpenStackDataPlaneSpec struct {
 
 	// +kubebuilder:validation:Optional
-	// Nodes - Map of nodes
-	Nodes map[string]OpenStackDataPlaneNodeSpec `json:"nodes,omitempty"`
-	// +kubebuilder:validation:Optional
-	// Roles - Map of roles
-	Roles map[string]OpenStackDataPlaneRoleSpec `json:"roles,omitempty"`
+	// NodeSets - Map of NodeSet names
```

Comment:
+1 for this being only the names not embeded.
im not sure we need the openstackDataplaneCR to exist however so i woudl be tempted to remvoe this file entirely instead